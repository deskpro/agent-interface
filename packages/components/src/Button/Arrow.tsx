import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/dp-Arrow.css";

export type ArrowProps = {
  isActive?: boolean;
  onClick?: (e: React.SyntheticEvent) => void;
};

export default React.forwardRef<HTMLSpanElement, ArrowProps>(
  ({ isActive, onClick }, ref) => (
    <span
      className={classNames("dp-Icon dp-Arrow", { "is-active": isActive })}
      ref={ref}
      role="button"
      tabIndex={-1}
      onClick={onClick}
      onKeyPress={e => e.key === "Enter" && onClick && onClick(e)}
    />
  )
);
