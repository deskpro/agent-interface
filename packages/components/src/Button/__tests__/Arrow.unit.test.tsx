import * as React from "react";
import { render, fireEvent } from "react-testing-library";

import Arrow from "../Arrow";

describe("<Arrow />", () => {
  it("Simple snapshot", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Arrow data-testid="arrow-button" onClick={handleClick} />
    );
    expect(getByTestId("arrow-button")).toMatchSnapshot();
  });
  it("should call onClick fn on click", () => {
    const handleClick = jest.fn();
    const { getByTestId } = render(
      <Arrow data-testid="arrow-button" onClick={handleClick} />
    );

    fireEvent(
      getByTestId("arrow-button"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true
      })
    );
    expect(handleClick.mock.calls.length).toBe(1);
  });
});
