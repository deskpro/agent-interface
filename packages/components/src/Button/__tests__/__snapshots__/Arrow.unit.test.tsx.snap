// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> should match its snapshot 1`] = `
<span
  class="dp-Icon dp-Arrow"
  data-testid="arrow-button"
  role="button"
  tabindex="-1"
/>
`;
